charts/prefect-server/templates/service.yaml

```diff
@@ -26,6 +26,7 @@ spec:
   {{- if eq .Values.service.type "NodePort" }}
   externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy | quote }}
   {{- end }}
+  {{- $serviceType := .Values.service.type }}
```

Comment:
I don't think this is neccessary?

Reply:
I need to define this variable to hold the value of serviceType. The variable is used inside the loop of `{{- range .Values.service.extraPorts }}` to set nodePort. While inside the loop, it can't access the .Values.service.type due to context being changed. Is there better way to handle this situation?

Reply:
ah i see - yes i think this makes sense!

Reply:
You should be able to access the value with `$.Values.service.type` (notice the `$.` instead of just `.`).

There's an example at the bottom of https://helm.sh/docs/chart_template_guide/variables.

Reply:
@mitchnielsen  Thank you for the tip. It works. $ - this variable will always point to the root context. Great to learn this.